elastic/kibana
# [Synthetics] Remove Saved Object Client usage from `use_recently_viewed_monitors` hook

Relates to #https://github.com/elastic/kibana/issues/153399

## Summary

Removes the deprecated Saved Object Client dependency from `useRecentlyViewedMonitors` hook.

-  [`getAllSyntheticsMonitorRoute`](https://github.com/elastic/kibana/pull/159033/files#diff-ee80010902693c9786b3043c6ec4e2f42a62fc33e55f9c7666bbf7034fe6d165R34
) will now accept additional filter `monitorQueryIds` which will only query the saved objects for the provided Monitor Query IDs.
- The PR also refactors the [`useMonitorName`](https://github.com/elastic/kibana/pull/159033/files#diff-c066d4869d55ec490d753a0febd925eeb12778836edd3603481128bdc51142feR30) to fix a bug with name duplication case on Monitor Add/Edit where the form was allowing the duplicate name to be saved. Also, since `useMonitorName` was using [`useMonitorList`](https://github.com/elastic/kibana/pull/159033/files#diff-f4f40e819338bf8c22f9c060e4941576f9659d3c7a206130884554b776d4b606R22) which wasn't returning the correct "loading" state for a query search due to `quietFetchMonitorListAction`, `useMonitorName` is refactored to utilize the `fetchMonitorManagementList` to have its own data and "loading" flag. The `query` parameter from `useMonitorList` is removed as `useMonitorName` was the only consumer.

### Testing
- The change impacts the following two UI components
   1.  Quick monitor selector dropdown on Synthetics -> Monitors -> _Monitor details page_ (the one right after the monitor's name heading, openable by `⌄` button).
   2. The monitor name duplicate validation on Monitor Add/Edit form.
- Create a bunch of monitors (should be > 10)
- Test while adding as well editing a monitor that the name duplicate validation works as expected
- Visit a few monitors by selecting via "Quick monitor selector dropdown" and confirm that it's behaving as expected
